pkg/server/desthost_backend_manager.go

```diff
@@ -36,19 +36,19 @@ func NewDestHostBackendManager() *DestHostBackendManager {
 }
 
 // Backend tries to get a backend associating to the request destination host.
-func (dibm *DestHostBackendManager) Backend(ctx context.Context) (Backend, error) {
+func (dibm *DestHostBackendManager) Backend(ctx context.Context) (Backend, string, error) {
 	dibm.mu.RLock()
 	defer dibm.mu.RUnlock()
 	if len(dibm.backends) == 0 {
-		return nil, &ErrNotFound{}
+		return nil, "", &ErrNotFound{}
 	}
 	destHost := ctx.Value(destHost).(string)
 	if destHost != "" {
 		bes, exist := dibm.backends[destHost]
 		if exist && len(bes) > 0 {
 			klog.V(5).InfoS("Get the backend through the DestHostBackendManager", "destHost", destHost)
-			return dibm.backends[destHost][0], nil
+			return dibm.backends[destHost][0], destHost, nil
```

Comment:
Using the agentID as the backend identifier seems reasonable. If we do that I think we should always make it what we return. agentID and destHost are not easily comparable but either the system or anyone examining the logs.